README.md

```diff
@@ -1,196 +1,289 @@
+- ```{"input_name1": "value1", "input_name2": "value2"}```
+Example of a properly formatted `downstream_input` for a `workflow`
+- ```{"0.step0_input": "value1", "1.step2_input": "value2"})```
+*Note the numerical index prefix necessary when specifying input for an `workflow`, which disambiguates which step in the workflow an input is targeted to*
+#### Files generated
+We use a hypothetical example of a local RUN folder named `20160101_M000001_0001_000000000-ABCDE`, that was placed into the `monitored_directory`, after the `dx-streaming-upload` role has been set up.
+**Local Files Generated**
+```
+path/to/LOG/directory
+(specified in monitor_run_config.template file)
+- 20160101_M000001_0001_000000000-ABCDE.lane.all.log
+path/to/TMP/directory
+(specified in monitor_run_config.template file)
+- no persistent files (tar files stored transiently, deleted upon successful upload to DNAnexus)
+```
+**Files Streamed to DNAnexus project**
+```
+project
+   └───20160101_M000001_0001_000000000-ABCDE
+       │───runs
+       │    │  RunInfo.xml
+       │    │  SampleSheet.csv
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all.log
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all.upload_sentinel
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all_000.tar.gz
+       │    │  run.20160101_M000001_0001_000000000-ABCDE.lane.all_001.tar.gz
+       │    │  ...
+       │
+       └───reads (or analyses)
+            │  output files from downstream applet (e.g. demx)
+            │   "reads" folder will be created if an applet is triggered
+            │   "analyses" folder will be created if a workflow is triggered
+            │  ...
+```
+The `reads` folder (and subfolders) will only be created if `applet` is specified.
+The `analyses` folder (and subfolder) will only be created if `workflow` is specified.
+`RunInfo.xml` and `SampleSheet.csv` will only be upladed if they can be located within the root of the local RUN directory.
+#### Logging, Notification and Error Handling
+**Uploading**
+A log of the CRON command (executed with `bash -e`) is written to the user's home folder `~/dx-stream_cron.log` and can be used to check the top level command triggered.
+The verbose log of the upload process (generated by the top-level `monitor_runs.py`) is written to the user's home folder `~/monitor.log`.
+These logs can be used to diagnose failures of upload from the local machine to DNAnexus.
+**Downstream applet**
+The downstream applet will be run in the project that the RUN directory is uploaded to (as specified in role variable `upload_project`). Users can log in to their DNAnexus account (corresponding to the `dx_token` or `dx_user_token`) and navigate to the upload project to monitor the progress of the applet triggered. Typically, on failure of a DNAnexus job, the user will receive a notification email, which will direct the user to check the log of the failed job for further diagnosis and debugging.

```

Comment:
--> " The downstream applet will be run in the project that the RUN directory is uploaded to (as specified in role variable `upload_project`). Users can log in to their DNAnexus account (corresponding to the `dx_token` or `dx_user_token` provided) and navigate to the upload project to monitor the progress of the applet triggered. Typically, upon failure of a DNAnexus job, the user will receive a notification email directing them to check the log of the failed job for further diagnosis and debugging. "